src/configs/config.ts

```diff
@@ -49,27 +48,40 @@ export class AutoblocksConfig<T> {
     this._value = value;
   }
 
-  private makeRequestUrl(args: { id: string; version: ConfigVersion }): string {
-    const configId = encodeURIComponent(args.id);
-    let version: string;
-    if (
-      args.version === RevisionSpecialVersionsEnum.DANGEROUSLY_USE_UNDEPLOYED
-    ) {
-      version = REVISION_UNDEPLOYED_VERSION;
-    } else {
-      version = encodeURIComponent(args.version);
+  private isRemoteConfigRefreshable(args: { remoteConfig: RemoteConfig }) {
+    return (
+      'latest' in args.remoteConfig ||
+      ('dangerouslyUseUndeployed' in args.remoteConfig &&
+        'latest' in args.remoteConfig.dangerouslyUseUndeployed)
+    );
+  }
+
+  private makeRequestUrl(args: { remoteConfig: RemoteConfig }): string {
+    const configId = encodeURIComponent(args.remoteConfig.id);
+    const base = `${API_ENDPOINT}/configs/${configId}`;
+    // Overide the version using the revisionId in the map if it exists
+    const revisionId = configRevisionsMap()[args.remoteConfig.id];
```

Comment:
in the prompt SDK i'm being super defensive about the overrides and double checking in multiple places that we're in a testing context:

* https://github.com/autoblocksai/javascript-sdk/blob/a499c2114d099e1d77c351088af1bd88a3729057/src/prompts/manager.ts#L316
* https://github.com/autoblocksai/javascript-sdk/blob/a499c2114d099e1d77c351088af1bd88a3729057/src/prompts/manager.ts#L372

should we do the same here? so just:

```ts
if (isTestingContext() && configRevisionsMap()[args.remoteConfig.id]) {
  const revisionId = configRevisionsMap()[args.remoteConfig.id];
  return `${base}/revisions/${revisionId}`;
}
```

just b/c it would be easy to accidentally refactor something like `configRevisionsMap()` to return the revision ID even when not in a testing context or something

but maybe it's overly defensive, lmk

Reply:
ya I think that makes sense, I update this check for testing context